src/WSConn.cpp

```diff
@@ -66,6 +67,17 @@ std::string WSConn::recv(void)
   // cout<<"RESULT:"<<endl;
   // cout<<result<<endl;
   // cout<<"END RESULT"<<endl;
+
+  if (meta->flags & CURLWS_CLOSE) {
+    uint16_t reason = 0;
+
+    if (result.size() >= 2) {
```

Comment:
should this not be `result.size() == 2` ?

I'm wondering if we shouldn't just hexdump `result` as reason?

Reply:
https://en.wikipedia.org/wiki/WebSocket#Opcodes - it is 2 bytes for the integer, plus optional additional bytes for a string. Printing that string would be cool, but HA does not currently put one in. The `>` just prepares us (without printing more info) for when it one day might.

Reply:
I see now that while we handle the "somebody might add a string later" case, we don't handle the " The payload is optional" case. Both could be better :)